### PR TITLE
feat(command): add \d command to show table column info

### PR DESCRIPTION
**Pull request by sandflee**
Signed-off-by: sandflee [sandflee.fang@gmail.com](mailto:sandflee.fang@gmail.com)

```
\d pg_catalog.contributors
+---+-----------+--------+----------+---+
| 0 | github_id | string | not null |   |
+---+-----------+--------+----------+---+

\d nation
+---+-------------+--------+----------+---+
| 0 | n_nationkey | int    | not null |   |
| 1 | n_name      | string | not null |   |
| 2 | n_regionkey | int    | not null |   |
| 3 | n_comment   | string | nullable |   |
+---+-------------+--------+----------+---+
```

### CODE REVIEW

File: src/catalog/root.rs [hunk 1]
```diff
@@ -59,6 +59,16 @@ impl RootCatalog {
...
     }
 
+    pub fn get_table_by_name(&self, name: &str) -> Option<Arc<TableCatalog>> {
+        let name = name.to_lowercase();
+        let (schema_name, table_name) = split_name(&name)?;
+        let ref_id = self.get_table_id_by_name(schema_name, table_name);
+        match ref_id {
+            Some(ref_id) => self.get_table(&ref_id),
+            None => None,
+        }
```
Comment (wangrunji0408) [inline src/catalog/root.rs:69]:
```suggestion
        let table_ref_id = self.get_table_id_by_name(schema_name, table_name)?;
        self.get_table(&table_ref_id)
```

**Review by wangrunji0408 — COMMENTED** [review]
Generally LGTM. Thanks for your contribution!

File: src/db.rs [hunk 2]
```diff
@@ -59,6 +59,48 @@ impl Database {
...
+        let mut column_name = Utf8ArrayBuilder::new();
+        let mut column_type = Utf8ArrayBuilder::new();
+        let mut column_is_null = Utf8ArrayBuilder::new();
+        let mut column_is_primary = Utf8ArrayBuilder::new();
+        let table_catalog = self.catalog.get_table_by_name(table_name).unwrap();
+
+        let all_columns = table_catalog.all_columns();
+        for (id, column) in &all_columns {
+            let name = column.name();
+            let data_type = column.datatype().kind().to_string().to_ascii_lowercase();
```
[the remark this answers is not in the document]
Comment (sandflee) [reply src/db.rs:73]: I could create a table using `create table t(a int, b string)`, if `\d` shows `integer/varchar` not `int/string`, is this the expected behaviour?    `standard name of data type` is the pg standard ?

[the remark this answers is not in the document]
Comment (wangrunji0408) [reply src/db.rs:73]: Yes, for `standard name of data type` I meant the pg standard.
But now I think you are right. RisingLight doesn't strictly follow pg so showing `int/string` is totally okay.
Sorry for my bothering.

**Review by wangrunji0408 — APPROVED** [review]
LGTM

**Pull request merged**